src/script_opt/Expr.cc

```diff
@@ -2223,14 +2268,21 @@ StmtPtr ListExpr::ReduceToSingletons(Reducer* c) {
     return red_stmt;
 }
 
+bool ListExpr::AllConstantOps() const {
```

Comment:
Doesn't seem  overly optimization specific, so could go in `src/Expr.cc`, too.

Reply:
Done!